versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5775 @@
+}
+
+BamAndIndex b_and_i = BamAndIndex { 
+  bam: "NA12878.bam",
+  bam_index: "NA12878.bam.bai" 
+}
+```
+
+#### Optional Types and None
+
+A type may have a `?` postfix quantifier, which means that it is an optional type, and its value is allowed to be undefined. It can only be used in calls or functions that accept optional values.
+
+WDL has a special value `None` whose meaning is "an undefined value". The type of the `None` value is `Any`, meaning it can be assigned to an optional declaration of any type. The `None` value is the only value that can be of type `Any`.
+
+An optional declaration has a default initialization of `None`, which indicates that it is undefined. An optional declaration may be initialized to any literal or expression of the correct type, including the special `None` value.
+
+```wdl
+Int certainly_five = 5      # an non-optional declaration
+Int? maybe_five_and_is = 5  # a defined optional declaration
+
+# the following are equivalent undefined optional declarations
+String? maybe_five_but_is_not
+String? maybe_five_but_is_not = None
+
+Boolean test_defined = defined(maybe_five_but_is_not) # Evaluates to false
+Boolean test_defined2 = defined(maybe_five_and_is)    # Evaluates to true
+Boolean test_is_none = maybe_five_but_is_not == None  # Evaluates to true
+Boolean test_not_none = maybe_five_but_is_not != None # Evaluates to false
+```
+
+For more details, see the section on [Optional Inputs with Defaults](#optional-inputs-with-defaults).
+
+#### Type Conversion
+
+WDL has some limited facilities for converting a value of one type to another type. Some of these are explicitly provided by [standard library](#standard-library) functions, while others are [implicit](#type-coercion). When converting between types, it is best to be explicit whenever possible, even if an implicit conversion is allowed.
+
+The execution engine is also responsible for converting (or "serializing") input values when constructing commands, as well as "deserializing" command outputs. For more information, see the [Command Section](#command-section) and the more extensive Appendix on [WDL Value Serialization and Deserialization](#appendix-a-wdl-value-serialization-and-deserialization).
+
+##### Type Coercion
+
+There are some pairs of WDL types for which there is an obvious, unambiguous conversion from one to the other. In these cases, WDL provides an automatic conversion (called "coercion") from one type to the other, such that a value of one type can be used anywhere the other type is expected.
+
+For example, file paths are always represented as strings, making the conversion from `String` to `File` obvious and unambiguous.
+
+```wdl
+String path = "/path/to/file"
+File f = path
+```
+
+The table below lists all globally valid coercions. The "target" type is the type being coerced to (this is often called the "left-hand side" or "LHS" of the coercion) and the "source" type is the type being coerced from (the "right-hand side" or "RHS").
```

Comment:
Move line 608 to here.  

"The table below lists all globally valid coercions, beyond the coercion of all primitive types which can always be converted to `String` using [string interpolation](#expression-placeholders-and-string-interpolation). The "target" type is the type being coerced to (this is often called the "left-hand side" or "LHS" of the coercion) and the "source" type is the type being coerced from (the "right-hand side" or "RHS")."

Reply:
Actually I don't think that section belongs in type coercion at all since it's a conversion, not a coercion. I'm moved it to its own section.